pkg/epp/scheduling/framework/plugins/filter/decision_tree_filter.go

```diff
@@ -47,6 +55,80 @@ type DecisionTreeFilter struct {
 	NextOnSuccessOrFailure framework.Filter
 }
 
+type decisionTreeFilterParameters struct {
+	Current                *decisionTreeFilterEntry `json:"current"`
+	NextOnSuccess          *decisionTreeFilterEntry `json:"nextOnSuccess"`
+	NextOnFailure          *decisionTreeFilterEntry `json:"nextOnFailure"`
+	NextOnSuccessOrFailure *decisionTreeFilterEntry `json:"nextOnSuccessOrFailure"`
+}
+
+type decisionTreeFilterEntry struct {
+	PluginRef    *string                       `json:"pluginRef"`
+	DecisionTree *decisionTreeFilterParameters `json:"decisionTree"`
+}
+
+func DecisionTreeFilterFactory(name string, rawParameters json.RawMessage, handle plugins.Handle) (plugins.Plugin, error) {
+	parameters := decisionTreeFilterParameters{}
+	if err := json.Unmarshal(rawParameters, &parameters); err != nil {
+		return nil, fmt.Errorf("failed to parse the parameters of the '%s' filter - %w", name, err)
+	}
+	return loadDecisionTree(&parameters, handle)
+}
+
+func loadDecisionTree(parameters *decisionTreeFilterParameters, handle plugins.Handle) (framework.Filter, error) {
+	result := &DecisionTreeFilter{}
+	var err error
+
+	result.Current, err = loadDecisionTreeEntry(parameters.Current, handle)
+	if err != nil {
+		return nil, err
+	}
+	if result.Current == nil {
+		return nil, errors.New("a current filter must be specified")
+	}
+
+	result.NextOnSuccess, err = loadDecisionTreeEntry(parameters.NextOnSuccess, handle)
+	if err != nil {
+		return nil, err
+	}
+
+	result.NextOnFailure, err = loadDecisionTreeEntry(parameters.NextOnFailure, handle)
+	if err != nil {
+		return nil, err
+	}
+
+	result.NextOnSuccessOrFailure, err = loadDecisionTreeEntry(parameters.NextOnSuccessOrFailure, handle)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func loadDecisionTreeEntry(entry *decisionTreeFilterEntry, handle plugins.Handle) (framework.Filter, error) {
+	if entry == nil {
+		return nil, nil
+	}
+
+	if entry.PluginRef != nil && entry.DecisionTree != nil {
+		return nil, errors.New("both pluginRef and decisionTree may not be specified")
+	}
+
+	if entry.PluginRef != nil {
+		instance := handle.Plugins().Plugin(*entry.PluginRef)
+		if instance == nil {
```

Comment:
This is unrelated but I think the `handle` should provide a `GetPlugin()` method and return an error if the plugin ref doesn't exist. That way is more explicit and less error prone.

Reply:
++, agree with one small difference. 
I suggest GetPlugin to return (plugin, bool) same as map does (bool - if key was found or not)

Reply:
There was a discussion about having Plugin() return (plugins.Plugin, bool), but was decided to leave it as is.

In general getters in Go do not have the prefix `Get` unlike for example Java.